platform/jewel/int-ui/int-ui-standalone/src/main/kotlin/org/jetbrains/jewel/intui/standalone/component/JDialogRenderer.kt

```diff
@@ -0,0 +1,305 @@
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.currentCompositionLocalContext
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberUpdatedState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.awt.ComposePanel
+import androidx.compose.ui.focus.FocusDirection
+import androidx.compose.ui.input.key.Key
+import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.input.key.KeyEventType
+import androidx.compose.ui.input.key.key
+import androidx.compose.ui.input.key.onKeyEvent
+import androidx.compose.ui.input.key.onPreviewKeyEvent
+import androidx.compose.ui.input.key.type
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.MeasurePolicy
+import androidx.compose.ui.layout.MeasureResult
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.layout.positionOnScreen
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.semantics.popup
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntRect
+import androidx.compose.ui.unit.IntSize
+import androidx.compose.ui.unit.round
+import androidx.compose.ui.util.fastForEach
+import androidx.compose.ui.util.fastMap
+import androidx.compose.ui.util.fastMaxBy
+import androidx.compose.ui.util.fastRoundToInt
+import androidx.compose.ui.window.PopupPositionProvider
+import androidx.compose.ui.window.PopupProperties
+import java.awt.AWTEvent
+import java.awt.Rectangle
+import java.awt.Toolkit
+import java.awt.Window
+import java.awt.event.AWTEventListener
+import java.awt.event.MouseEvent
+import java.awt.event.WindowEvent
+import java.lang.ref.WeakReference
+import javax.swing.JDialog
+import org.jetbrains.jewel.foundation.modifier.thenIf
+import org.jetbrains.jewel.ui.component.PopupRenderer
+
+/**
+ * A popup renderer implementation that uses JDialog to display popups in a Compose UI.
+ *
+ * This renderer creates a JDialog that hosts Compose content, allowing for integration between Swing and Compose UI
+ * hierarchies.
+ *
+ * Note that this renderer requires you to enable the 'compose.interop.blending' system property. Otherwise, the popup
+ * will be rendered with a white background. For more information, check the official documentation available at
+ * https://www.jetbrains.com/help/kotlin-multiplatform-dev/compose-desktop-swing-interoperability.html#experimental-interop-blending
+ *
+ * @param window The parent window that will own the popup dialog
+ */
+public class JDialogRenderer(window: Window) : PopupRenderer {
+    private val currentWindow = WeakReference(window)
```

Comment:
I'm not sure if the weak ref here is really needed. I still need to review. Probably can be removed as the instance should be released on the compose dispose